anchors/dna/anchors/anchors.js

```diff
@@ -1,528 +1,99 @@
-
-// script globals
-var ANCHOR_GENERIC = "_anchor_generic_";
-var ENTRY_TYPE_PROP_NAME = "__ANCHORS_ENTRY_TYPE__";
-var ANCHOR_DELIMETER = "_-_anchor_delimeter_-_";
+/**
+ * @param anchor Object of the form {anchorType: <string>, anchorText: <string>}
+ * @return anchorHash
+ **/
+function anchor(anchor){
+  var anchorType = {anchorType: anchor.anchorType, anchorText: ''};
+  var rootAnchortype =  {anchorType: 'anchorTypes', anchorText: ''};
+  var anchorHash = makeHash('anchor', anchor);
+  var anchorGet = get(anchorHash);
+  if(anchorGet === null){
+    var anchorType = {anchorType: anchor.anchorType, anchorText: ''};
+    var rootAnchortype =  {anchorType: 'anchorTypes', anchorText: ''};
+    var anchorTypeGet = get(makeHash('anchor', anchorType));
+    if(anchorTypeGet === null){
+      var rootAnchorTypeHash = makeHash('anchor', rootAnchortype);
+      if (get(rootAnchorTypeHash) === null){
+        rootAnchorTypeHash = commit('anchor', rootAnchortype);
+      }
+      var anchorTypeHash = commit('anchor', anchorType);
+      commit('anchor_link', { Links:[{Base: rootAnchorTypeHash, Link: anchorTypeHash, Tag: anchorType.anchorType}]});
+    } else {
+      anchorTypeHash = makeHash('anchor', anchorType);
+    }
+    anchorHash = commit('anchor', anchor);
+    commit('anchor_link',  { Links:[{Base: anchorTypeHash, Link: anchorHash, Tag: anchor.anchorText}]});
+  }
```

Comment:
I wonder if we might consider refactoring this logic into a helper function to reduce repetition.